src/prefect/environments/storage/docker.py

```diff
@@ -533,59 +550,40 @@ def create_dockerfile_object(self, directory: str) -> str:
         with open(healthcheck_loc, "w") as health_file:
             health_file.write(healthcheck)
 
-        if self.dockerfile:
-            with open(self.dockerfile, "r") as contents:
-                base_commands = textwrap.indent("\n" + contents.read(), prefix=" " * 16)
-        else:
-            base_commands = "FROM {base_image}".format(base_image=self.base_image)
+        # Escape the healthcheck location
+        healthcheck_loc = (
+            healthcheck_loc.replace("\\", "/") if self.dockerfile else "healthcheck.py"
+        )
+
+        # Generate the command to run the healthcheck
+        healthcheck_run = ""
+        if not self.ignore_healthchecks:
+            flow_file_paths = ", ".join(['"{}"'.format(k) for k in self.flows.values()])
+            python_version = (sys.version_info.major, sys.version_info.minor)
+            healthcheck_run = (
+                f"RUN python {self.prefect_directory}/healthcheck.py "
+                f"'[{flow_file_paths}]' '{python_version}'"
+            )
 
         file_contents = textwrap.dedent(
-            """\
-            {base_commands}
+            f"""
+            {multiline_indent(base_commands, 12)}
```

Comment:
Here and below, use a f-string instead of a format call with a bunch of repeated variables and use `multiline_indent` to indent multiline strings rather than indenting with confusing numbers above